### PR TITLE
Add ability for custom dnsConfig to multiple charts

### PR DESCRIPTION
**Pull request by stafot**
#### What this PR does / why we need it:
Adding possibility to set a custom dnsConfig for the pods. Pod's DNS Config field enable users more control on the DNS settings for the pods.
#### Which issue this PR fixes
Adds custom dnsConfig value to the following charts:

- alertmanager
- kube-prometheus-stack
- prometheus
- prometheus-adapter
- prometheus-elasticsearch-exporter
- prometheus-node-exporter

#### Special notes for your reviewer:
https://github.com/prometheus-community/helm-charts/issues/634
#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by bismarck** [comment]
@stafot I believe you're going to have to create a PR per chart?

**Comment by caarlos0** [comment]
> @stafot I believe you're going to have to create a PR per chart?

yes, pretty sure.

File: charts/alertmanager/Chart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ icon: https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/a
 sources:
   - https://github.com/prometheus/alertmanager
 type: application
-version: 0.5.0
+version: 0.5.1
```
Comment (desaintmartin) [inline charts/alertmanager/Chart.yaml:9]: Please bump minor version instead of patch: this is a new feature.

**Review by desaintmartin — COMMENTED** [review]
Thanks! I understand that there is going to be several PRs instead of this one.

**Comment by stafot** [comment]
Ok i will split is and bump versions as instructed. Thanks!

**Pull request closed without merging**